Emby.Server.Implementations/Library/DefaultPasswordResetProvider.cs

```diff
@@ -73,7 +82,14 @@ public async Task<PinRedeemResult> RedeemPasswordResetPin(string pin)
 
         public async Task<ForgotPasswordResult> StartForgotPasswordProcess(MediaBrowser.Controller.Entities.User user, bool isInNetwork)
         {
-            string pin = new Random().Next(99999999).ToString("00000000",CultureInfo.InvariantCulture);
+            string pin = string.Empty;
+            using (var cryptoRandom = System.Security.Cryptography.RandomNumberGenerator.Create())
+            {
+                byte[] bytes = new byte[4];
+                cryptoRandom.GetBytes(bytes);
+                pin = bytes.ToString();

```

Comment:
which encoding it uses, hex?

Reply:
https://dotnetfiddle.net/ACkwGM It seems to write "System.Byte[]" at least in vanilla .NET

Using BitConverter seems to work. And the UI can add the dashes while writing, reading something with the dashes might be easier for people.

Reply:
Fixed